src/shacl-shape-lib/elements-shacl-shape.xsl

```diff
@@ -17,102 +17,134 @@
     <xsl:import href="../common/utils.xsl"/>
     <xsl:import href="../common/formatters.xsl"/>
     <xsl:import href="../common/checkers.xsl"/>
+    <xsl:import href="descriptors-shacl-shape.xsl"/>
 
     <xsl:output method="xml" encoding="UTF-8" byte-order-mark="no" indent="yes"
         cdata-section-elements="lines"/>
 
 
     <xd:doc>
-        <xd:desc>[Rule 2] - (Class in data shape layer). Specify declaration axiom for UML Class as
-            SHACL Node Shape where the URI and a label are deterministically generated from the
-            class name and rules 3,7,8 </xd:desc>
+        <xd:desc>Rule C.02. Class — in data shape layer. Specify a SHACL NodeShape declaration axiom 
+            for each UML Class. The URIs of the node shape and of the OWL class they refer to, 
+            are deterministically generated from the UML Class name. </xd:desc>
     </xd:doc>
     <xsl:template match="element[@xmi:type = 'uml:Class']">
         <xsl:variable name="class" select="."/>
+        <xsl:variable name="shapeClassUri"
+            select="fn:concat($base-shape-uri, $defaultDelimiter, replace($class/@name, ':', '-'))"/>
```

Comment:
no concats and no replacement

Reply:
this is the third version of the same logic:
first you were providing variables with concat/replace; then you were doing in-line replace/concat; no you do a combined replace & concat. This is not a professional code. This is a hasty code to finish it ASAP. Write professional code please, it will be faster. Trust me @Dragos0000 .